src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,192 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(null, value, null);
+        if (first == null) {
+            first = newNode;
+            last = newNode;
+        } else {
+            last.next = newNode;
+            newNode.prev = last;
+            last = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndex(index);
+        Node<T> newNode = new Node<>(null, value, null);
+        if (index == 0) {
+            if (first == null) {
+                first = newNode;
+                last = newNode;
+            } else {
+                newNode.next = first;
+                first.prev = newNode;
+                first = newNode;
+            }
+        } else if (index == size) {
+            if (last == null) {
+                first = newNode;
+                last = newNode;
+            } else {
+                newNode.prev = last;
+                last.next = newNode;
+                last = newNode;
+            }
+        } else {
+            Node<T> current = first;
+            for (int i = 0; i < index; i++) {
+                current = current.next;
+            }
+            Node<T> prevNode = current.prev;
+            newNode.next = current;
+            newNode.prev = prevNode;
+            prevNode.next = newNode;
+            current.prev = newNode;
+        }
+        size++;
     }
```

Comment:
The logic for finding the node at a specific index is repeated; consider creating a 'findNodeByIndex' method. Also, the code for inserting a node is repeated; consider creating an 'insertBefore' method to reduce duplication.